packages/@lwc/ssr-compiler/src/compile-template/transformers/component.ts

```diff
@@ -29,13 +31,31 @@ const bYieldFromChildGenerator = esTemplateWithYield`
     {
         const childProps = ${is.objectExpression};
         const childAttrs = ${is.objectExpression};
-        async function* childSlottedContentGenerator() {
-            ${is.statement}
+        const slottedContent = {
+            light: Object.create(null),
+            shadow: async function* () {
+                ${/* shadow slot content */ is.statement}
+            }
         };
-        yield* ${is.identifier}(${is.literal}, childProps, childAttrs, childSlottedContentGenerator);
+        function addContent(name, fn) {
+            let contentList = slottedContent.light[name]
+            if (contentList) {
+                contentList.push(fn)
+            } else {
+                slottedContent.light[name] = [fn]
+            }
+        }
+        ${/* addContent statements */ is.callExpression}
+        yield* ${is.identifier}(${is.literal}, childProps, childAttrs, slottedContent);
```

Comment:
Future optimization: dedupe content by changing `slottedContent.shadow` to be an array and add `slottedContent.shadow.push(fn)` in `addContent`. This will require updating the content of the generated functions to conditionally render the `slot` attribute.